components/Ripple/src/MDCRippleTouchController.h

```diff
@@ -0,0 +1,184 @@
+ and logic to provide the correct ripple effect based on the user interacting with the view the
+ ripple is added to.
+ */
+@interface MDCRippleTouchController : NSObject <UIGestureRecognizerDelegate>
+
+/**
+ A weak reference to the view that the ripple is added to, and that responds to the user
+ events.
+ */
+@property(nonatomic, weak, readonly, nullable) UIView *view;
+
+/**
+ The ripple view added to the view.
+ */
+@property(nonatomic, strong, readonly, nonnull) MDCRippleView *rippleView;
+
+/** Delegate to extend the behavior of the touch control. */
+
+/**
+ A delegate to extend the behavior of the touch controller.
+ */
+@property(nonatomic, weak, nullable) id<MDCRippleTouchControllerDelegate> delegate;
+
+/** Gesture recognizer used to bind touch events to ripple. */
+
+/**
+ The gesture recognizer used to bind the touch events to the ripple.
+ */
+@property(nonatomic, strong, readonly, nonnull) UILongPressGestureRecognizer *gestureRecognizer;
+
+/**
+ The selection gesture recognizer used to bind the touch events related to selection to the ripple.
+ */
+@property(nonatomic, strong, readonly, nullable)
+    UILongPressGestureRecognizer *selectionGestureRecognizer;
+
+/**
+ This BOOL tells the touch controller if to allow selection and all the logic and visuals
+ that come with it, for this ripple.
+
+ Defaults to NO.
+ */
+@property(nonatomic) BOOL allowsSelection;
+
+/**
+ This BOOL is set to YES if the ripple is currently selected, or NO otherwise.
+ It only has significance if selectionMode is activated.
+
+ Defaults to NO.
+ */
+@property(nonatomic, getter=isSelected) BOOL selected;
+
+/**
+ This BOOL is set to YES if the ripple is currently in the selection mode, or NO otherwise.
+
+ Defaults to NO.
+ */
+@property(nonatomic) BOOL selectionMode;
+
+/**
+ Sets the color of the ripple for state.
+
+ @param rippleColor The ripple color to set the ripple to.
+ @param state The state of the ripple in which to set the ripple color.
+ */
+- (void)setRippleColor:(nullable UIColor *)rippleColor forState:(MDCRippleState)state;
+
+/**
+ Gets the ripple color for the given state.
+
+ @param state The ripple's state.
+ @return the color of the ripple for state.
+ */
+- (nullable UIColor *)rippleColorForState:(MDCRippleState)state;
+
+/**
+ Sets the alpha of the ripple for state.
+
+ @param rippleAlpha The ripple alpha to set the ripple to.
+ @param state The state of the ripple in which to set the ripple alpha.
+ */
+- (void)setRippleAlpha:(CGFloat)rippleAlpha forState:(MDCRippleState)state;
+
+/**
+ Gets the ripple alpha for the given state.
+
+ @param state The ripple's state.
+ @return the alpha of the ripple for state.
+ */
+- (CGFloat)rippleAlphaForState:(MDCRippleState)state;
+
+/**
+ The current state of the ripple.
+ */
+@property(nonatomic, readonly) MDCRippleState state;
+
+/** Unavailable, please use initWithView: instead. */
+
+/**
+ Unavailable, please use `initWithView` instead.
+ */
+- (nonnull instancetype)init NS_UNAVAILABLE;
```

Comment:
_Tip:_ Shouldn't we move the initializers above property declaration for readability?